notebook/notebook/notebook.py

```diff
@@ -55,10 +55,7 @@
             workshops = await cursor.fetchall()
 
             if len(workshops) != 1:
-                set_message(
-                    session,
-                    'You are not logged into an active workshop.  Please log in to join the workshop.',
-                    'error')
+                del session['workshop_session']
```

Comment:
I think we should communicate something back to the user. For 0, "We couldn't find your workshop. Please log in to try again". For 1, "Oops! You seem to be logged in to multiple workshops, please try logging in again". Or so.

Reply:
There are only two options, 0 or 1.  Only 0 is an error here.

So I initially did that with the message above and it felt confusing in testing.  I imagine this will occur in two cases:

 - Someone leaves a workshop and tries to use it again after is deactivated.
 - Someone comes to another workshop with cookies left over from a previous one.

There shouldn't be a message in the second case.  Given the disclaimer on the workshop index page, I felt OK with no message on the first one.  In general, I think I prefer less feedback to confusing feedback.

Reply:
If it's only 0/1 I think it would be clearer to check for 0.

Reply:
Hmm, I prefer this since it is safer.  `name` is a unique key, so the query can only return one value, but I'd rather not give the user access to a notebook if something is clearly wrong than give the user access to a random notebook that they potentially shouldn't have access to.  (I mean, they should have access to both, but it's hard to figure out how this could happen so I'm just speculating here.)

If you'd want some additional documentation, I can put `assert not notebooks` or `assert len(notebooks) == 0` inside the if and we'll get an internal error in that (impossible) case.

Reply:
Not a big deal. I would prefer we check for all 3 possibilities, so that we can log the strange case (or 2+), however you would like to do that. Not blocking

Reply:
I think the argument is a bit inconsistent. If it’s only possible to get 0 or 1, then checking 0 explicitly provides added code clarity. If either is possible, but 1+ represents a programmatic error that is difficult to catch during testing, I think we should explicitly check that condition (so check 0, 1+ in separate if statements), and alert our developers that this happened so they can fix the condition, because this condition is quite important (since users are expected to have only 1 or 0, 1+ means either were being hacked, or something has happened that will potentially lead to a user not getting access to the notebook they expect, namely in this case access to no notebooks).